testing/tests/DevExpress.ui.widgets.editors/dropDownBox.tests.js

```diff
@@ -493,21 +492,55 @@ QUnit.module('popup options', moduleConfig, () => {
 
         try {
             instance.open();
-
             this.clock.tick();
-            const popup = $('.dx-popup').dxPopup('instance');
+
+            const popup = this.$element.find('.dx-popup').dxPopup('instance');
             const maxHeight = popup.option('maxHeight');
+            assert.roughEqual(Math.floor(maxHeight()), startPopupHeight, 3, 'maxHeight is correct');
+        } finally {
+            scrollTop.restore();
+            this.$element.css('margin-top', 0);
+        }
+    });
+
+    QUnit.test('maxHeight should be distance between the popup top bound and the element top bound if the popup has been rendered at the top already and the window was scrolled (T874949, T942217)', function(assert) {
+        const scrollTopValue = 50;
+        let startPopupHeight;
+        this.$element.dxDropDownBox({
+            width: 300,
+            contentTemplate: (e) => {
+                const content = $('<div id=\'dd-content\'></div>');
+
+                setTimeout(() => {
```

Comment:
Maybe it's better to get overlay content height after DDB opening below to avoid using `setTimeout`. Or are some problems there?